module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -333,24 +341,29 @@ namespace module::input {
                     .count(),
                 0.0);
 
-            // Assume support foot change has occurred if walk state has changed
-            if (current_walk_phase != walk_state.phase) {
-                current_walk_phase = walk_state.phase;
+            // If our planted foot phase has changed from a single planted foot to another single planted foot, update
+            // anchor frame
+            if (current_planted_foot_phase != sensors->planted_foot_phase
+                && sensors->planted_foot_phase != WalkState::Phase::DOUBLE) {
+                current_planted_foot_phase = sensors->planted_foot_phase;
                 // Compute the new anchor frame (Hwp) (new support foot)
-                if (current_walk_phase.value == WalkState::Phase::LEFT) {
+                if (current_planted_foot_phase.value == WalkState::Phase::LEFT) {
                     Hwp = Hwp * sensors->Htx[FrameID::R_FOOT_BASE].inverse() * sensors->Htx[FrameID::L_FOOT_BASE];
+                    // Set the z translation, roll and pitch of the anchor frame to 0 as assumed to be on field plane
+                    Hwp.translation().z() = 0;
+                    Hwp.linear() = rpy_intrinsic_to_mat(Eigen::Vector3d(0, 0, mat_to_rpy_intrinsic(Hwp.linear()).z()));
                 }
-                else if (current_walk_phase.value == WalkState::Phase::RIGHT) {
+                else if (current_planted_foot_phase.value == WalkState::Phase::RIGHT) {
                     Hwp = Hwp * sensors->Htx[FrameID::L_FOOT_BASE].inverse() * sensors->Htx[FrameID::R_FOOT_BASE];
+                    // Set the z translation, roll and pitch of the anchor frame to 0 as assumed to be on field plane
+                    Hwp.translation().z() = 0;
+                    Hwp.linear() = rpy_intrinsic_to_mat(Eigen::Vector3d(0, 0, mat_to_rpy_intrinsic(Hwp.linear()).z()));
```

Comment:
Would it be possible to turn this to a switch case instead? Given that we already use enum values set up, using switch should hopefully make it much faster